src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const newStudents: Student[] = [...students];
+
+  const calcAvg: (arr: number[])=> number = (arr) => {
+    return arr.reduce((sum, grade) => sum + (grade / arr.length), 0);
+  };
+
+  newStudents.sort((a: Student, b: Student) => {
```

Comment:
you can not add types here as well, TS already know about typeof of this variables
```suggestion
  newStudents.sort((a, b) => {
```